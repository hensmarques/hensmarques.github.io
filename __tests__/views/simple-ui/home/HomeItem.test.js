import React from 'react';
import HomeItem from '../../../../src/views/simple-ui/home/HomeItem';

describe('<HomeItem/>', () => {
  it('renders without crashing', () => {
    shallow(<HomeItem />);
  });

  it('renders the props correctly and matches the snapshot', () => {
    const mockProps = {
      header: 'mockHeader',
      content: 'mock content',
      icon: (
        <svg>
          <circle />
        </svg>
      ),
      action: <button>mock action</button>
    };
    const wrapper = shallow(<HomeItem {...mockProps} />);
    expect(
      wrapper.containsAllMatchingElements([
        <h4>{mockProps.header}</h4>,
        <div className="icon-inner">{mockProps.icon}</div>,
        <div className="item-content">
          <p>{mockProps.content}</p>
          <div>{mockProps.action}</div>
        </div>
      ])
    ).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
