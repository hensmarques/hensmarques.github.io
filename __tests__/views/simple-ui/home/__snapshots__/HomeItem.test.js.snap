// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeItem/> renders the props correctly and matches the snapshot 1`] = `
<div
  className="item"
>
  <h4>
    mockHeader
  </h4>
  <div
    className="item-container"
  >
    <div
      className="icon"
    >
      <div
        className="icon-inner"
      >
        <svg>
          <circle />
        </svg>
      </div>
    </div>
    <div
      className="item-content"
    >
      <p>
        mock content
      </p>
      <div>
        <button>
          mock action
        </button>
      </div>
    </div>
    <div
      className="check"
    >
      <div
        className="check-inner"
      >
        <CheckIcon />
      </div>
    </div>
  </div>
</div>
`;
